[TT-13186/TT-13275/TT-13331] document upstream basic authentication (#5566)
TykTechnologies/tyk-docs: tyk-docs/content/basic-config-and-security/security/upstream-authentication/basic-authentication.md

---
title: Upstream Basic Authentication
tags: ["upstream-basic-auth"]
description: "How to authenticate upstream requests with basic authentication"
menu:
  main:
    parent: "Upstream Authentication"
weight: 2
---

If your upstream API is protected with basic authentication, you can configure Tyk to send requests with basic authentication credentials.

- You can specifiy username and password to be used. 
- You can configure the header in which basic authenticaiton credentials are to be sent, the default header to be used is `Authorization`. 

## How To Set Up

### Via API Definition

> Inside your OAS API definition you should configure `x-tyk-api-gateway.upstream.authentication.basicAuth` field.

Reviewer: ```suggestion
In your OAS API definition, configure the `x-tyk-api-gateway.upstream.authentication.basicAuth` field, which is an object containing the following properties:
```

- `enabled` needs to be true to enable upstream basic authentication.

> - `headerName` is the header to be used.

Reviewer: ```suggestion
- `headerName` is the header to be used. Defaults to `Authorization` if not specified.
```

- `username` is the username to be used.
- `password` is the password to be used.

{{< note success >}}
**Note**

> `x-tyk-api-gateway.upstream.authentication.enabled` needs to be true to enable upstream authentication.

Reviewer: ```suggestion
`x-tyk-api-gateway.upstream.authentication. basicAuth.enabled` needs to be true to enable upstream authentication.
```

> Inside your OAS API definition you should configure `x-tyk-api-gateway.upstream.authentication.basicAuth` field.
My understanding is that this field holds a JSON object with 4 values.

Author: This is a prerequisite, `x-tyk-api-gateway.upstream.authentication.basicAuth.enabled` is already mentioned. 
`x-tyk-api-gateway.upstream.authentication.enabled` decides if upstream authentication needs to be performed at all.
